evalml/tests/component_tests/test_simple_imputer.py

```diff
@@ -269,38 +269,55 @@ def test_simple_imputer_with_none():
     transformed = imputer.transform(X, y)
     expected = pd.DataFrame({"int with None": [1, 0, 5, 2],
                              "float with None": [0.1, 0.0, 0.5, 0.2]})
-    assert_frame_equal(expected, transformed.to_dataframe(), check_dtype=False)
+    assert_frame_equal(expected, transformed, check_dtype=False)
 
     X = pd.DataFrame({"category with None": pd.Series(["b", "a", "a", None], dtype='category'),
-                      "boolean with None": pd.Series([True, None, False, True], dtype='boolean'),
+                      "boolean with None": pd.Series([True, None, False, True]),
                       "object with None": ["b", "a", "a", None],
                       "all None": [None, None, None, None]})
     y = pd.Series([0, 0, 1, 0, 1])
     imputer = SimpleImputer()
     imputer.fit(X, y)
     transformed = imputer.transform(X, y)
     expected = pd.DataFrame({"category with None": pd.Series(["b", "a", "a", "a"], dtype='category'),
-                             "boolean with None": pd.Series([True, True, False, True], dtype='boolean'),
+                             "boolean with None": pd.Series([True, True, False, True], dtype='category'),
                              "object with None": pd.Series(["b", "a", "a", "a"], dtype='category')})
-    assert_frame_equal(expected, transformed.to_dataframe(), check_dtype=False)
+    assert_frame_equal(expected, transformed, check_dtype=False)
+
+
+def test_simple_imputer_supports_natural_language():
+    X = pd.DataFrame({"cat with None": ["a", "b", "a", None],
+                      "natural language col": ["free-form text", "will", "be imputed", None]})
+    y = pd.Series([0, 0, 1, 0, 1])
+    X.ww.init(logical_types={"natural language col": "NaturalLanguage"})
+    imputer = SimpleImputer(impute_strategy="constant", fill_value="placeholder")
```

Comment:
👍 
Could call this test `test_simple_imputer_supports_natural_language_constant`